src/service/impl/trips/converters.ts

```diff
@@ -0,0 +1,219 @@
+    authority: undefined,
+    fromEstimatedCall: fromEstimatedCall,
+    fromPlace: mapToLegacyPlace(leg.fromPlace),
+    toPlace: mapToLegacyPlace(leg.toPlace),
+    interchangeTo: undefined,
+    intermediateEstimatedCalls: intermediateEstimatedCalls,
+    line: mapToLegacyLine(leg.line),
+    mode: mapToLegacyLegMode(leg.mode),
+    pointsOnLink: mapToPointsOnLink(leg.pointsOnLink),
+    // Casting to allow servicejourney to be null
+    serviceJourney: mapToLegacyServiceJourney(
+      leg.serviceJourney
+    ) as ServiceJourney,
+    situations: [],
+    transportSubmode: mapToLegacySubMode(leg.transportSubmode)
+  };
+}
+
+type PointsOnLink_v2 = Leg_v2['pointsOnLink'];
+function mapToPointsOnLink(points: PointsOnLink_v2): PointsOnLink | undefined {
+  if (!points || !points.points || !points.length) return;
+  return {
+    length: points.length,
+    points: points.points
+  };
+}
+
+type EstimatedCall_v2 = Leg_v2['fromEstimatedCall'];
+function mapToLegacyEstimatedCall(
+  estimatedCall: EstimatedCall_v2,
+  serviceJourney: ServiceJourney_v2,
+  frontText: string
+): EstimatedCall | undefined {
+  if (!estimatedCall) return;
+
+  return {
+    ...estimatedCall,
+    aimedArrivalTime: '',
+    expectedArrivalTime: '',
+    cancellation: false,
+    date: '',
+    forAlighting: true,
+    forBoarding: true,
+    predictionInaccurate: false,
+    realtime: true,
+    requestStop: false,
+    // Casting to allow servicejourney to be null
+    serviceJourney: mapToLegacyServiceJourney(serviceJourney) as ServiceJourney,
+    destinationDisplay: {
+      frontText: frontText
+    },
+    notices: [],
+    quay: undefined
+  };
+}
+
+type IntermediateEstimatedCall_v2 = Leg_v2['intermediateEstimatedCalls'][0];
+function mapToLegacyIntermediateEstimatedCall(
+  estimatedCall: IntermediateEstimatedCall_v2,
+  serviceJourney: ServiceJourney_v2,
+  frontText: string
+): EstimatedCall | undefined {
+  if (!estimatedCall) return;
+
+  return {
+    quay: undefined,
+    date: estimatedCall.date,
+    aimedArrivalTime: '',
+    expectedArrivalTime: '',
+    aimedDepartureTime: '',
+    expectedDepartureTime: '',
+    cancellation: false,
+    forAlighting: true,
+    forBoarding: true,
+    predictionInaccurate: false,
+    realtime: true,
+    requestStop: false,
+    // Casting to allow servicejourney to be null
+    serviceJourney: mapToLegacyServiceJourney(serviceJourney) as ServiceJourney,
+    destinationDisplay: {
+      frontText: frontText
+    },
+    notices: []
+  };
+}
+
+type Place_v2 = Leg_v2['fromPlace'];
+function mapToLegacyPlace(place: Place_v2): Place {
+  return {
+    ...place,
+    name: place.name || '',
+    quay: mapToLegacyQuay(place.quay)
+  };
+}
+
+type Line_v2 = Leg_v2['line'];
+function mapToLegacyLine(line: Line_v2): Line | undefined {
+  if (!line) return;
+  return {
+    ...line,
+    transportMode: TransportMode.BUS, // Not used in app
+    transportSubmode: TransportSubmode.LOCAL, // Not used in app
```

Comment:
Maybe use `Unknown` instead as transport mode and sub mode for the situations where we don't have a value or don't use the value. If we want to fallback to transport mode bus we should use `localBus` as sub mode, as sub mode `local` is for trains.

Reply:
`Unknown` isn't an option in the SDK type that's being returned here. Setting to `localBus` instead.